api/v1alpha1/bmcjob_types.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2022 Tinkerbell.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// BMCJobConditionType represents the condition of the BMC Job.
+type BMCJobConditionType string
+
+const (
+	// JobCompleted represents successful completion of the BMC Job tasks.
+	JobCompleted BMCJobConditionType = "Completed"
+	// JobFailed represents failure in BMC job execution.
+	JobFailed BMCJobConditionType = "Failed"
+	// JobTaskCompleted represents successful completion of a BMC Job task.
+	JobTaskCompleted BMCJobConditionType = "TaskCompleted"
+)
+
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// BMCJobSpec defines the desired state of BMCJob
+type BMCJobSpec struct {
+	// Tasks represents a list of baseboard management actions to be executed.
+	// The tasks are completed in order.
+	Tasks []Task `json:"tasks"`
+}
+
+// Task represents the action to be performed.
+// A single task can only perform one type of action.
+// For example either PowerAction or BootAction.
+// +kubebuilder:validation:MaxProperties:=1
+type Task struct {
+	// PowerAction represents a baseboard management power operation.
+	PowerAction *PowerAction `json:"powerAction,omitempty"`
+
+	// BootAction represents a baseboard management boot device operation.
+	BootAction *BootAction `json:"bootAction,omitempty"`
+}
+
+type PowerAction struct {
+	// State represents the requested power state to set for the baseboard management.
+	// +kubebuilder:validation:Enum=On;Off;Status;Cycle;Reset;Soft
+	State PowerState `json:"state"`
+}
+
+type BootAction struct {
```

Comment:
```suggestion
type BootDeviceAction struct {
```

`BootAction` is too ambiguous.